### PR TITLE
hack: verify-go-version.sh in all the repo

### PR DESCRIPTION
**Pull request by sttts**
So many places where we reference a go version.

Check that the build go version matches.

### CODE REVIEW

**Review by stevekuznetsov — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -249,6 +249,10 @@ verify-k8s-deps:
 verify-imports:
 	hack/verify-imports.sh
 
+.PHONY: verify-go-imports
+verify-go-versions:
+	hack/verify-go-versions.sh
```
Comment (stevekuznetsov) [inline Makefile:254]:
```suggestion
	hack/verify-go-version.sh
```

Comment (sttts) [reply Makefile:254]: yes, fixed

**Comment by sttts** [comment]
@stevekuznetsov with 1.18 most image builds fail here with a killed linker. Do we need more memory?

**Pull request merged**